src/index.ts

```diff
@@ -243,12 +243,41 @@ const transformObsoleteColorStops = (args: Array<string>): string[] => {
  * ```
  */
 export function fromCSS(str: string) {
-    const values = str.match(/([a-zA-Z0-9_-]+)\((.*)\).*/);
+    if (str.endsWith(";")) {
+        str = str.slice(0, -1);
+    }
+    const values = str?.match(/([a-zA-Z0-9_-]+)\((.*)\).*/);
     if (!values) throw new Error("Invalid CSS Gradient function: " + str);
 
     const [, method, argString] = values;
-    const args = argString.split(",").map((v) => v.trim());
+    const args: string[] = [];
+
+    let buffer = "";
+    let openParens = 0;
+
+    for (let i = 0; i < argString.length; i++) {
```

Comment:
Could maybe come up with a better way to do this than looping over every character if needed.